go-controller/pkg/node/node.go

```diff
@@ -330,7 +330,22 @@ func deleteServiceConntrackEntries(svc *kapi.Service) {
 		}
 		err := deleteConntrack(svc.Spec.ClusterIP, port.Port, port.Protocol)
 		if err != nil {
-			klog.Errorf("Failed to delete conntrack entry for %s:%d: %v", svc.Spec.ClusterIP, port.Port, err)
+			klog.Errorf("Failed to delete conntrack entry for ClusterIP %s:%d: %v", svc.Spec.ClusterIP, port.Port, err)
+		}
+		// delete the conntrack entries for nodePorts
+		if port.NodePort > 0 {
+			localAddrSet, err := getLocalAddrs()
```

Comment:
This does not need to be initialized for every port. You can do it at the top of the method if it's a `NodePort` service.

Reply:
yeah, good point